src/TLuaInterpreter.cpp

```diff
@@ -80,6 +82,9 @@ int luaopen_yajl(lua_State*);
 
 using namespace std;
 
+QTextToSpeech* speechUnit;
```

Comment:
This is C++ not C! I think it ought to be possible to add a `QTextToSpeech* mpSpeechUnit` as a `private` member in the header file `.h`(suitably guarded with some sort of "do we have the speech module?" `#if define` or what have you) - set the __pointer__ in the right place in the initialisation list below starting at line 89 in the new code (items should be listed in the same order as they appear in the header file) to the null value:

    , mpSpeechUnit( nullptr )

then whenever this is used in another function it should be checked and initialised if needed:

    #if defined (YES_WE_HAVE_THE_SPEECH_MODULE)
        if (!mpSpeechUnit) {
            mpSpeechUnit = new QTextToSpeech();
            if (!mpSpeechUnit) {
                lua_pushnil(L);
                lua_pushstring(L, "Sorry, text-to-speech engine failed to initialise - Mudlet is currently speechless.")
            } else {
                 mpSpeechUnit->setVolume(1.0);
                 mpSpeechUnit->setRate(0.0);
                 mpSpeechUnit->setPitch(0.0);
            }
       }

    // .... stuff using the speech functions

    #endif

Obviously, you could pull the initialisation out to a separate method/function but things are slightly more complicated because the lua functions that are registered in `TLuaInterpreter::initLuaGlobals()` are static (global) functions that are not associated with a particular instance of the `TLuaInterpreter` class __and do not have a `this` member or access to an instance of the class's non `static` members__ {As a C++ newbie it took me a while to get that straight...! 😌 }